modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/TerraformBootDeployment.java

```diff
@@ -0,0 +1,203 @@
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot.api.TerraformApi;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot.model.TerraformAsyncDeployFromDirectoryRequest;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot.model.TerraformAsyncDestroyFromDirectoryRequest;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot.model.WebhookConfig;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot.model.WebhookConfig.AuthTypeEnum;
+import org.eclipse.xpanse.modules.deployment.utils.DeployEnvironments;
+import org.eclipse.xpanse.modules.models.service.common.enums.Csp;
+import org.eclipse.xpanse.modules.models.service.deploy.DeployResult;
+import org.eclipse.xpanse.modules.models.servicetemplate.Ocl;
+import org.eclipse.xpanse.modules.models.servicetemplate.enums.DeployerKind;
+import org.eclipse.xpanse.modules.orchestrator.deployment.DeployTask;
+import org.eclipse.xpanse.modules.orchestrator.deployment.DeployValidationResult;
+import org.eclipse.xpanse.modules.orchestrator.deployment.Deployment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of th deployment with terraform.
+ */
+@Slf4j
+@Profile("terraform-boot")
+@Component
+public class TerraformBootDeployment implements Deployment {
+
+    public static final String STATE_FILE_NAME = "terraform.tfstate";
+    public static final String TF_DEBUG_FLAG = "TF_LOG";
+    private final DeployEnvironments deployEnvironments;
+    private final TerraformVersionProvider terraformVersionProvider;
+    private final TerraformBootConfig terraformBootConfig;
+    private final TerraformLocalConfig terraformLocalConfig;
+
+    /**
+     * Initializes the Terraform deployer.
+     */
+    @Autowired
+    public TerraformBootDeployment(
+            DeployEnvironments deployEnvironments,
+            TerraformVersionProvider terraformVersionProvider,
+            TerraformBootConfig terraformBootConfig,
+            TerraformLocalConfig terraformLocalConfig) {
+        this.deployEnvironments = deployEnvironments;
+        this.terraformVersionProvider = terraformVersionProvider;
+        this.terraformBootConfig = terraformBootConfig;
+        this.terraformLocalConfig = terraformLocalConfig;
+    }
+
+    @Override
+    public DeployResult deploy(DeployTask deployTask) {
+        DeployResult result = new DeployResult();
+        TerraformApi terraformApi = new TerraformApi();
+        TerraformAsyncDeployFromDirectoryRequest request;
+        try {
+            request = getDeployRequest(deployTask);
+            terraformApi.asyncDeployWithScripts(request);
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
```

Comment:
we throw a proper exception back. We can define a new exception - 'TerraformBootRequestFailed'